sdk/appconfiguration/app-configuration/src/internal/tracingHelpers.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+  getTracer,
+  Span,
+  SpanOptions,
+  SpanKind,
+  CanonicalCode,
+} from "@azure/core-tracing";
+
+import {
+  RequestOptionsBase,
+  RestError
+} from "@azure/core-http";
+
+/**
+ * @internal
+ * @ignore
+ */
+export class Spanner<TClient> {
+  tracer = getTracer();
+
+  constructor(private baseOperationName: string, private componentName: string) {    
+  }
+
+  async trace<T>(
+    operationName: keyof TClient,
+    options: RequestOptionsBase,
+    fn: (span: Span, options: RequestOptionsBase) => Promise<T>,
+    translateToCanonicalCodeFn: (err: Error) => CanonicalCode = Spanner.getCanonicalCode    
+  ) : Promise<T> {
+    var { span, newOptions } = this.setupSpan(options, operationName);
+
+    try {
+      return await fn(span, newOptions);
+    } catch (err) {
+      span.setStatus({
+        code: translateToCanonicalCodeFn(err),
+        message: err.message
+      });
+      throw err;
+    } finally {
+      span.end();
+    }
+  }
+
+  private setupSpan(options: RequestOptionsBase, operationName: keyof TClient) {
+    const spanOptions: SpanOptions = {
```

Comment:
You'll also want to check if `options` has a parentSpan and set it in spanOptions. This will ensure that the span created in line 53 is part of the same trace as our parent span.

Reply:
It was  doing this but that cascading set of ...'s did make it unclear that it was happening. :)

I refactored the code a bit and added in a test that shows it working :)